### ISSUE TITLE
To many Ressources per block generated by Mining + pick-up bug

### ISSUE THREAD

**Issue by faiith** [StrangeLoopGames/EcoIssues#11790]
Version: Stagion 0.8.1.0 e7012ee4
Mods: None

In mining, too much material is generated for every block that has been mined. Normally you would get 4 resources per block, but in the Stagin version you get 10-15.

In addition (I do not have the skill for multiple pick up) is always trying to pick up everything on the ground. If many rocks are next to each other, so it goes beyond 20, you can not collect anything (it comes the message 'no space in the inventory')

**Comment by D3nnis3n** [comment]
Cannot reproduce, could you attach a video?

**Comment by faiith** [comment]
Sure, i'm at work now, but i'll attach a video as soon as i'm home

**Comment by faiith** [comment]
Well, can't attach video because i'm not able to connect to server anymore :/  We have installed on our server the latest staging version available on the official website: Staging - beta 8.1.0 # e7012ee4 Unfortunately, our clients can no longer connect to the server because the staging version that ships via Steam is already higher version number: #743897c95

**Comment by faiith** [comment]
Found a workarround for connection-issue... just overwrote gamefile in steam-folder with files from official website^^

So, here's the link to the requested video (youtube): https://youtu.be/9XE7jBMJzgc

In the VIdeo, i'm showing 

- i don't have the pick-up skill from mason/mining
- i'm mining 1 block, and getting 16 limestone
- all limestones where picked up
- after mining 3 blocks, it's not possible to pickup at all

**Comment by mirasrael** [comment]
Can you please attach your server save to this issue?

**Comment by mirasrael** [comment]
Also seems you're using older version. Can you test with Official Beta Server?

**Comment by faiith** [comment]
@misrael: How can attach the server-save here in github, game.db is 250MB large, game.eco 92MB, but here is maximum of 10MB allowed? Or are other files meant?

Yeah, wir using not newest staging version, because there is no server download with newest version available on your website.

**Comment by mirasrael** [comment]
You can get latest server version from Steam. 
Tools -> Eco Server [bleeding_edge]

**Comment by mirasrael** [comment]
You can upload the sever save (only *.eco needed) to Google Drive and put link here.

**Comment by faiith** [comment]
Link for *.eco: 

https://drive.google.com/file/d/1Xb4mHugLbFoeT0DEIayo4G7K3qAU9PNR/view?usp=sharing

**Comment by faiith** [comment]
Updated Server now on newest version, issue still occurs

![382310_20190328173305_1](https://user-images.githubusercontent.com/40213281/55175444-a015f200-517f-11e9-9c1a-87af72c72b5f.png)

**Comment by D3nnis3n** [comment]
Still cannot reproduce this, but found another bug, try to make a video please.

**Comment by faiith** [comment]
I did a video allready ;)

> So, here's the link to the requested video (youtube): https://youtu.be/9XE7jBMJzgc
> 
> In the VIdeo, i'm showing
> 
>     * i don't have the pick-up skill from mason/mining
> 
>     * i'm mining 1 block, and getting 16 limestone
> 
>     * all limestones where picked up
> 
>     * after mining 3 blocks, it's not possible to pickup at all

**Comment by D3nnis3n** [comment]
Mh, weird. Still can't reproduce this, but i can reproduce that rubble can't be picked up at all. We're going to take a look on this in playtest, maybe someone else can reproduce it.

**Comment by SeanAtr0n** [comment]
I'm seeing this behavior in-editor, along with this exception:
```
Server encountered an exception:

<size=60.00%>Exception: Exception
Message:A MultiAtomicAction failed in the middle of execution. Which is bad. Error message: Trying to remove non-existing block
Source:Eco.Core

System.Exception: A MultiAtomicAction failed in the middle of execution. Which is bad. Error message: Trying to remove non-existing block
   at Eco.Core.Utils.AtomicAction.MultiAtomicAction.TryApply() in C:\slg\Eco\Server\Eco.Core\Utils\AtomicAction\MultiAtomicAction.cs:line 57
   at Eco.Gameplay.Objects.RubbleObject`1.TryPickup(Player actor, Inventory inv) in C:\slg\Eco\Server\Eco.Gameplay\Objects\RubbleObject.cs:line 326
   at Eco.Gameplay.Objects.RubbleObject`1.OnActInteract(InteractionContext context) in C:\slg\Eco\Server\Eco.Gameplay\Objects\RubbleObject.cs:line 309
   at Eco.Gameplay.Players.Player.<>c__DisplayClass60_0.<PlayerInteract>b__2(IInteractable inter) in C:\slg\Eco\Server\Eco.Gameplay\Players\Player.cs:line 323
   at Eco.Gameplay.Interactions.InteractionExtensions.<>c__DisplayClass2_1`1.<ExecuteUntilSuccess>b__1() in C:\slg\Eco\Server\Eco.Gameplay\Interactions\IInteractable.cs:line 102
   at Eco.Gameplay.Interactions.InteractionExtensions.ExecuteUntilSuccess(IEnumerable`1 results) in C:\slg\Eco\Server\Eco.Gameplay\Interactions\IInteractable.cs:line 111
   at Eco.Gameplay.Interactions.InteractionExtensions.ExecuteUntilSuccess[T](IEnumerable`1 enumerable, Func`2 resultFunc) in C:\slg\Eco\Server\Eco.Gameplay\Interactions\IInteractable.cs:line 102
   at Eco.Gameplay.Players.Player.PlayerInteract(InteractionInfo info) in C:\slg\Eco\Server\Eco.Gameplay\Players\Player.cs:line 329</size>
```

**Comment by SeanAtr0n** [comment]
Bug requires multiple people having the sweeping hands talent on your server

**Comment by D3nnis3n** [comment]
That explains why it didnt occur for me

Sean Ater <notifications@github.com> schrieb am Do., 28. März 2019, 23:21:

**Comment by mirasrael** [comment]
@faiith and @SeanAtr0n Nice finding! Fixed in 8.1-7
